src/shared/components/visitor-counter/VisitorCounter.tsx

```diff
@@ -0,0 +1,46 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { getVisitorCount } from '../../../app/actions/getVisitorCount';
+
+const VisitorCounter = () => {
+  const [visitorCount, setVisitorCount] = useState<number | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchVisitorCount = async () => {
+      try {
+        const count = await getVisitorCount();
+        setVisitorCount(count);
+      } catch (err) {
+        setError('Failed to load visitor count. Please try again later.');
+      }
+    };
+
+    fetchVisitorCount();
+  }, []);
+
+  if (error) {
+    return (
+      <div className="fixed top-0 left-0 w-full text-center bg-red-500 py-2 text-lg font-bold text-white z-50 shadow-md">
+        ⚠️ {error}
```

Comment:
i would say to remove the error state on frontend, and just log it with console.error
User doesnt have to see the error state :)